tests/unit/utils_test/test_data_utils.py

```diff
@@ -22,14 +22,19 @@ def test_append_exception(self):
         with self.assertRaises(ValueError):
             append_data(data, 4)
 
+    def test_append_same_dim_array_arg(self):
+        data = np.array([[1, 2], [3, 4]])
+        arg = np.array([[5, 6]])
```

Comment:
`append_data` should be used to append a single element to an existing array. So, arg should be just `np.array([5, 6])`. np.append does not work in this case. Either you have to add a dimension to the arg, or just use np.vstack. I think vstack will be easier to understand.